### PR TITLE
update @clz and function callbacks for 0.10; make linenoiseX functions

### PR DESCRIPTION
**Pull request by garrisonhh**
It looks like linenoize doesn't compile with the current zig version, so I fixed some compilation errors.

I also made some functions public that seem like they should be, unsure if this is the intention of them originally

### CODE REVIEW

File: src/main.zig [hunk 1]
```diff
@@ -11,8 +11,8 @@ const enableRawMode = term.enableRawMode;
 const disableRawMode = term.disableRawMode;
 const getColumns = term.getColumns;
 
-pub const HintsCallback = (fn (Allocator, []const u8) Allocator.Error!?[]const u8);
-pub const CompletionsCallback = (fn (Allocator, []const u8) Allocator.Error![]const []const u8);
+pub const HintsCallback = *fn (Allocator, []const u8) Allocator.Error!?[]const u8;
+pub const CompletionsCallback = *fn (Allocator, []const u8) Allocator.Error![]const []const u8;
```
Comment (notramo) [inline src/main.zig:15]: The stage1 compiler requires the callbacks to be functions, not a function pointers.
These are referenced in line 152 and 227 in the `fun` variable. Currently, it is called like this:
```zig
try fun(self.allocator, self.buf.items);
```
Stage 1 can not dereference automatically, so a slight modification is required:
```zig
try fun.*(self.allocator, self.buf.items);
```
To support conditional compilation, depending on the compiler backend, it should be called like this:
```zig
const builtin = @import("builtin");
try (if (builtin.zig_backend == .stage1) fun.* else fun)(self.allocator, self.buf.items);
```

**Review by notramo — CHANGES_REQUESTED** [review]
Some modifications are required for making it compile on stage 1 also.
Unfortunately, runtime performance of stage 2 is 10< times worse in some cases (https://github.com/ziglang/zig/issues/12518), so it's necessary to use stage 1 in some projects, until stage 2 can produce the same runtime performance.

**Comment by garrisonhh** [comment]
I switched them to `std.meta.FnPtr` instead, and also fixed the analagous decls I found in `c.zig`!

**Comment by garrisonhh** [comment]
thanks for the catch :)

File: src/main.zig [hunk 2]
```diff
@@ -34,7 +34,7 @@ const key_ctrl_w = 23;
 const key_esc = 27;
 const key_backspace = 127;
 
-fn linenoiseEdit(ln: *Linenoise, in: File, out: File, prompt: []const u8) !?[]const u8 {
+pub fn linenoiseEdit(ln: *Linenoise, in: File, out: File, prompt: []const u8) !?[]const u8 {
```
Comment (joachimschmidt557) [inline src/main.zig:37]: These functions are intentionally not public as the API surface is intended to have only one entry point (`Linenoise.linenoise`).

**Review by joachimschmidt557 — COMMENTED** [review]
Thanks a lot for your contribution! Just one thing - it is intentional that only the `Linenoise.linenoise` function is the public API (similar to the original linenoise).

**Review by joachimschmidt557 — APPROVED** [review]

**Comment by joachimschmidt557** [comment]
Thanks!

**Pull request merged**